PTP does not compile with CONFIG_PTP_UDP_IPv6_PROTOCOL=y

### Describe the bug

When using UDPv6 for PTP, (setting CONFIG_PTP_UDP_IPv6_PROTOCOL=y) the mcast address of PTP is `s6_addr`.
However in [transport.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/ptp/transport.c#L198), the mcast address is still handled as ipv4 (`s_addr`). 


### Regression

- [ ] This is a regression.

### Steps to reproduce

1. set  CONFIG_PTP_UDP_IPv6_PROTOCOL=y 
2. west build samples/net/ptp

### Relevant log output

```shell
zephyr/subsys/net/lib/ptp/transport.c:201:72: error: 'struct in6_addr' has no member named 's_addr'; did you mean 's6_addr'?
  201 |                         net_sin(&m_addr)->sin_addr.s_addr = mcast_addr.s_addr;
```

### Impact

Functional Limitation – Some features not working as expected, but system usable.

### Environment

- OS: Arch Linux
- Toolchain: Zephyr SDK + openocd

### Additional Context

Addtionally, when compiling a hotfixed version, my STM32F767ZI will hang during boot.
No shell or other output provided.

Am I missing a device tree entry/overlay for the example or something else? The [sample ptp entry](https://docs.zephyrproject.org/latest/samples/net/ptp/README.html#ptp) says it is supported board. 
However, I can not find any ptp_clock entry in the dtsi in comparison to other hardware like the nxp frdm_k64f.

I am still debugging the issue and can try to provide more debug output soon, but I thought maybe someone else has a solution for that already.

--- a/subsys/net/lib/ptp/Kconfig
+++ b/subsys/net/lib/ptp/Kconfig
@@ -91,10 +91,12 @@ choice
 
 	config PTP_UDP_IPv4_PROTOCOL
 		bool "UDP with IPv4"
+		depends on NET_IPV4
 		select NET_IPV4_IGMP
 
 	config PTP_UDP_IPv6_PROTOCOL
 		bool "UDP with IPv6"
+		depends on NET_IPV6
 		select NET_IPV6_MLD
 endchoice
 

--- a/subsys/net/lib/ptp/transport.c
+++ b/subsys/net/lib/ptp/transport.c
@@ -15,11 +15,18 @@ LOG_MODULE_REGISTER(ptp_transport, CONFIG_PTP_LOG_LEVEL);
 
 #define INTERFACE_NAME_LEN (32)
 
+union mcast_addr {
+	struct in_addr ipv4;
+#if defined(CONFIG_PTP_UDP_IPv6_PROTOCOL)
+	struct in6_addr ipv6;
+#endif
+};
+
 #if CONFIG_PTP_UDP_IPv4_PROTOCOL
-static struct in_addr mcast_addr = {{{224, 0, 1, 129}}};
+static union mcast_addr mcast_addr = {.ipv4 = {{{224, 0, 1, 129}}}};
 #elif CONFIG_PTP_UDP_IPv6_PROTOCOL
-static struct in6_addr mcast_addr = {{{0xff, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
-				       0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x81}}};
+static union mcast_addr mcast_addr = {.ipv6 = {{{0xff, 0xe, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0,
+						 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1, 0x81}}}};
 #else
 #error "Chosen PTP transport protocol not implemented"
 #endif
@@ -198,7 +205,7 @@ static int transport_send(int socket, int port, void *buf, int length, struct so
 		if (IS_ENABLED(CONFIG_PTP_UDP_IPv4_PROTOCOL)) {
 			m_addr.sa_family = AF_INET;
 			net_sin(&m_addr)->sin_port = htons(port);
-			net_sin(&m_addr)->sin_addr.s_addr = mcast_addr.s_addr;
+			net_sin(&m_addr)->sin_addr.s_addr = mcast_addr.ipv4.s_addr;
 
 		} else if (IS_ENABLED(CONFIG_PTP_UDP_IPv6_PROTOCOL)) {
 			m_addr.sa_family = AF_INET6;

